### PR TITLE
Mishal filter diaries

### PR DESCRIPTION
**Pull request by misusonu18**
# added the filter
## video file

https://user-images.githubusercontent.com/52093984/160238447-7b615529-c0bd-4a03-bf2a-919dd73811d9.mp4

### CODE REVIEW

File: diaries.html [hunk 1]
```diff
@@ -383,1008 +409,62 @@ <h1>Freshbits Diaries</h1>
...
+                </li>
+
+                <li>
+                    <a class="btn blue" onclick="displayFilterSelection('workshop')">
+                        Workshops
+                    </a>
+                </li>
+
+                <li>
+                    <a class="btn pink" onclick="displayFilterSelection('fun')">
```
Comment (hirenkeraliya) [inline diaries.html:434]:
```diff
-displayFilterSelection('fun')
+displayActivityFor('fun')
```

File: diaries.html [hunk 2]
```diff
@@ -1435,6 +515,19 @@ <h4>Memories</h4>
     <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jQuery.Marquee/1.5.0/jquery.marquee.js"></script>
     <script type="text/javascript" src="js/lity.js"></script>
     <script type="text/javascript" src="//buttons.github.io/buttons.js" async defer></script>
+    <script type="text/javascript" src="js/data.js"></script>
```
Comment (hirenkeraliya) [inline diaries.html:518]: `diaries-data.js`

File: diaries.html [hunk 3]
```diff
@@ -1435,6 +515,19 @@ <h4>Memories</h4>
...
+        let filterUl = document.getElementById("filter-ul");
+        let buttonLi = filterUl.getElementsByClassName("btn");
+
+        for (let i = 0; i < buttonLi.length; i++) {
+            buttonLi[i].addEventListener("click", function() {
+            let current = document.getElementsByClassName("active-filter-button");
+            current[0].className = current[0].className.replace(" active-filter-button", "");
+            this.className += " active-filter-button";
+        });
+        }
```
Comment (hirenkeraliya) [inline diaries.html:530]: code is not needed. I think it's bad to call 2 listeners for the click-on button.

File: js/custom_script.js [hunk 4]
```diff
@@ -254,4 +254,53 @@ $(function () {
...
+                let imagesAndLinkDiv = linkAndImageTemplate.content.cloneNode(true);
+                imagesAndLinkDiv.querySelector('.image').href = image.media;
+                imagesAndLinkDiv.querySelector('.thumbnail-image').src = image.thumbnail;
+                imagesAndLinkDiv.querySelector('.thumbnail-image').alt = dairyData.title;
+                imageDiv.append(imagesAndLinkDiv);
+            });
+            return;
+        }
+
+        else if (filterName === 'all') {
```
Comment (hirenkeraliya) [inline js/custom_script.js:285]:
```javascript
if (condition) {
    
} else if(condition) {
    
}
```

A conditional statement can be written like this.

File: js/custom_script.js [hunk 5]
```diff
@@ -254,4 +254,53 @@ $(function () {
...
+            let dairiesCard = templateDairy.content.cloneNode(true);
+            let linkAndImageTemplate = dairiesCard.querySelector('#images-and-link');
+            let imageDiv = dairiesCard.querySelector('#add-images');
+            dairiesCard.querySelector('.freshbits-diaries-row').id = dairyData.id;
+            dairiesCard.querySelector('.event-date').innerHTML = dairyData.date;
+            dairiesCard.querySelector('.event-date').href = "#" + dairyData.id;
+            dairiesCard.querySelector('.card-title').innerHTML = dairyData.title;
+            dairiesCard.querySelector('.card-topic').innerHTML = dairyData.topic;
+            freshbitsDairiesDiv.append(dairiesCard);
+            dairyData.images.forEach((image) => {
```
Comment (hirenkeraliya) [inline js/custom_script.js:295]: make some spacing divide code if needed.

File: js/custom_script.js [hunk 6]
```diff
@@ -254,4 +254,53 @@ $(function () {
...
+                imagesAndLinkDiv.querySelector('.image').href = image.media;
+                imagesAndLinkDiv.querySelector('.thumbnail-image').src = image.thumbnail;
+                imagesAndLinkDiv.querySelector('.thumbnail-image').alt = dairyData.title;
+                imageDiv.append(imagesAndLinkDiv);
+            });
+        }
+    });
+}
+
+displayFilterSelection();
```
Comment (hirenkeraliya) [inline js/custom_script.js:306]: always all the files have a newline at end of the file.

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: js/custom_script.js [hunk 7]
```diff
@@ -254,4 +254,62 @@ $(function () {
...
+let freshbitsDairiesDiv = document.getElementById('freshbits-dairies-div');
+diariesData.reverse();
+function displayActivityFor(filterName = 'all', currentElement) {
+    let currentActiveClass = document.getElementsByClassName("active-filter-button");
+    currentActiveClass[0].className = currentActiveClass[0].className.replace(" active-filter-button", "");
+    currentElement.className += " active-filter-button";
+
+    freshbitsDairiesDiv.innerHTML = "";
+    diariesData.forEach((dairyData) => {
+        if (dairyData.tag === filterName) {
```
Comment (gauravmak) [inline js/custom_script.js:269]: Same code being repeated in else if

**Review by gauravmak — CHANGES_REQUESTED** [review]

**Pull request merged**